test/end-to-end/core.sh

```diff
@@ -84,193 +85,194 @@ wait_for_registry
 # services can end up on any IP.  Make sure we get the IP we need for the docker registry
 DOCKER_REGISTRY=$(oc get --output-version=v1beta3 --template="{{ .spec.portalIP }}:{{ with index .spec.ports 0 }}{{ .port }}{{ end }}" service docker-registry)
 
-registry="$(dig @${API_HOST} "docker-registry.default.svc.cluster.local." +short A | head -n 1)"
-[[ -n "${registry}" && "${registry}:5000" == "${DOCKER_REGISTRY}" ]]
+os::cmd::expect_success_and_text "dig @${API_HOST} docker-registry.default.svc.cluster.local. +short A | head -n 1 | xargs printf '%s:5000'" "${DOCKER_REGISTRY}"
 
 echo "[INFO] Verifying the docker-registry is up at ${DOCKER_REGISTRY}"
-wait_for_url_timed "http://${DOCKER_REGISTRY}/" "[INFO] Docker registry says: " $((2*TIME_MIN))
+os::cmd::try_until_success "curl --max-time 2 -fs http://${DOCKER_REGISTRY}/healthz" $((2*TIME_MIN))
```

Comment:
@deads2k, @liggitt and some others had this sort of conversation on another pull and we agreed that making wrapper methods that delegate to `os::cmd` is probably a bad idea, as it obfuscates what is going on. Please point out to me where `wait_for_url` does certificate things, as I did not see that. The only behavior I saw in the function was to run `curl` with some arguments over and over again, which this refactor makes much more obvious by pulling the arguments into the line.